packages/portal-admin/src/components/utils/index.ts

```diff
@@ -0,0 +1,16 @@
+export async function countdown (seconds: number, callback) {
+    return new Promise((resolve) => {
+      let remainingSeconds = seconds;
+  
+      const intervalId = setInterval(() => {
+        callback(remainingSeconds);
+  
+        if (remainingSeconds <= 0) {
+          clearInterval(intervalId);
```

Comment:
Does this piece of code work?

Referencing `intervalId` here

Reply:
Yes it does, even that it is defined on line 5 interval ID is still accessible inside the callback function and is used to terminate the running interval since we only want it to be running a set number of times